WalletWasabi.Gui/Controls/WalletExplorer/SendControlViewModel.cs

```diff
@@ -832,16 +836,26 @@ private void SetFeeTargetLimits()
 			}
 		}
 
-		private bool TryParseUserFee(out decimal userFee)
+		public bool TryParseUserFee(out decimal userFeeRateAsDecimal) => TryParseUserFeeCore(UserFeeText, out userFeeRateAsDecimal);
+
+		public static bool TryParseUserFeeCore(string feeText, out decimal userFeeRateAsDecimal)
 		{
-			userFee = default;
-			var userFeeText = UserFeeText;
-			return
-				userFeeText is { }
-				&& !userFeeText.Contains(",")
-				&& decimal.TryParse(userFeeText, out userFee)
-				&& (userFee * 1_000) < Constants.MaxSatoshisSupply
-				&& userFee > 0;
+			try
```

Comment:
Don't use exceptions for control flow control, just check the range is ok. try something like this:

```c#
	private bool TryParseUserFee(out decimal userFee) =>
		decimal.TryParse(UserFeeText, NumberStyles.Integer, new CultureInfo("en-US"), out userFee)
			&& (userFee < (decimal.MaxValue / 1_000)) 
			&& new FeeRate(userFee) is var userFeeRate
			&& userFeeRate > MinRelayTxFee && userFeeRate <= AbsurdlyHighFee;
```

Reply:
No, I need try/catch because even when you compare large decimals it triggers OverflowExceptions still. I just tried this one and it threw exceptions still.

Reply:
Ah it was the FeeRate constructor that throws up

Reply:
Ah it was the FeeRate constructor that throws up